src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -145,6 +140,17 @@ public SemanticModel(Compilation compilation, BicepSourceFile sourceFile, IFileR
             });
         }
 
+        private IEnumerable<ExportMetadata> ExportedTypes => Root.TypeDeclarations
```

Comment:
It's a bit of a blurry line, but I generally try to stick to the convention of:
* property getter -> extremely low cost, no performance surprises - basically "call it as often as you want without a care in the world".
* method -> may require some understanding of the perf characteristics before deciding to use it.

I'm not sure which category this example falls into, just thought I'd put it out there!